fenix/app/src/main/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckState.kt

```diff
@@ -21,8 +22,16 @@ sealed interface ReviewQualityCheckState : State {
 
     /**
      * The state when the user has not opted in for the feature.
+     *
+     * @property retailers List of retailer names to be displayed in order in the onboarding UI.
      */
-    object NotOptedIn : ReviewQualityCheckState
+    data class NotOptedIn(
+        val retailers: List<Int> = listOf(
+            R.string.review_quality_check_retailer_name_amazon,
```

Comment:
We could define the enums like you did for `ProductVendor`s and have them here, then in UI layer we can map them to string resources, keeping this State independent of `R`